### ISSUE TITLE
Progressbar is not showing while uploading audio or video to cloudinary server.

### ISSUE THREAD

**Issue by kaprat** [cloudinary/cloudinary_android#56]
I have shown progressbar using below way but progressbar is not showing.

```
ProgressDialog prd = new ProgressDialog(MainActivity.this);
prd.setTitle("Loading...");
prd.setMessage("Please, wait a moment");
prd.show();

Map uploadResult;
Map config = new HashMap();
config.put("cloud_name", "mycloudname");
config.put("api_key", "myapikey");
config.put("api_secret", "myapisecret");
Cloudinary cloudinary = new Cloudinary(config);

try {
    uploadResult = cloudinary.uploader().uploadLarge(videoName, ObjectUtils.asMap("resource_type", "video"));
    Log.d("VideoUpload", uploadResult.toString());
    if (prd != null && prd.isShowing()) {
        prd.dismiss();
    }
} catch (IOException e) {
    e.printStackTrace();
    uploadResult = null;
    if (prd != null && prd.isShowing()) {
        prd.dismiss();
    }
} catch (Exception e) {
    e.printStackTrace();
    uploadResult = null;
    if (prd != null && prd.isShowing()) {
        prd.dismiss();
    }
}
```

**Comment by yakirp** [comment]
Hi @kaprat, 

First, please make sure to use our l[atest Android SDK version](https://cloudinary.com/documentation/android_integration), which includes a new way to upload resources which you can track upload progress by getting callbacks on the following events: onStart, onProgress, onSuccess, onError, and onReschedule.

Per your question - you asked about why you can't see the progress bar on your UI?
If yes, I'm afraid this is not because of our code...

Best, 
Yakir

**Comment by kaprat** [comment]
Yes sir i am using the latest versions of libraries and also i have tried by uploading using the latest version code (Using MediaManager) but i am getting the issue that i have also reported you can check it from [here ](https://github.com/cloudinary/cloudinary_android/issues/55)

**Comment by kaprat** [comment]
Is it necessary to show progressDialog in onStart method of media manager and need to dismiss in onSuccess method of media manager ?

Can you help me how can i upload multiple image using media manager ?

**Comment by yakirp** [comment]
Hi @kaprat,

>Is it necessary to show progressDialog in onStart method of media manager and need to dismiss in onSuccess method of media manager ?

This is not necessary but should be great in terms of user experience.

>Can you help me how can i upload multiple image using media manager ?

You can just invoke the upload call with the file you need to upload, there is no need to wait for the current upload to finish. 

For example:

```
String[] files = { "images/1.jpg","images/2.jpg","images/3.jpg" };

        for (String path: files) {

            String requestId = MediaManager.get().upload(path).callback(new UploadCallback() {
                @Override
                public void onStart(String requestId) {

                }
                @Override
                public void onProgress(String requestId, long bytes, long totalBytes) {

                }
                @Override
                public void onSuccess(String requestId, Map resultData) {

                }
                @Override
                public void onError(String requestId, ErrorInfo error) {

                }
                @Override
                public void onReschedule(String requestId, ErrorInfo error) {

                }}).unsigned("unsigned").dispatch();

        }
```

**Comment by kaprat** [comment]
Thanks for reply sir.

This will upload all images and i'm able to receive onSuccess callback only once ?

**Comment by yakirp** [comment]
No, you will get the `onSuccess` event (like any other event) for **each** file, it's up to you to manage this and to check that all your files upload successfully, or failed. 

For example, you tried to upload 3 files, 2 files uploaded, but 1 has failed -in this case, you will get 2 `onSuccess`, and 1 `onError`.

**Comment by yakirp** [comment]
>Can you tell me where should i show and hide progressbar ?

If the progress is per file, so you can show it on `onStart`, and hide on `onSuccess`.

**Comment by kaprat** [comment]
Ok let me try.
Thanks Sir.

**Comment by roeeba** [comment]
Closing this issue due to the time elapsed. Please feel free to either re-open the issue, contact our support at http://support.cloudinary.com or create a new ticket if you have any additional issues.
